admin.php

```diff
@@ -1,45 +1,131 @@
  *
  * If any bugs occur, please submit a new issue
  * @see https://github.com/findologic/dokuwiki-plugin-findologic-xml-export/issues/new
  * @author Dominik Brader <support@findologic.com>
  */
 
-// must be run within Dokuwiki
-if(!defined('DOKU_INC')) {
-    define('DOKU_INC',realpath(dirname(__FILE__).'/../../../').'/');
+if (!defined('DOKU_INC')) {
+    die('Must be run within DokuWiki!');
 }
 
-class admin_plugin_findologicxmlexport extends DokuWiki_Admin_Plugin {
+require_once(__DIR__ . '/vendor/autoload.php');
+require_once(__DIR__ . '/PageGetter.php');
+require_once(__DIR__ . '/DokuwikiPage.php');
+
+class admin_plugin_findologicxmlexport extends DokuWiki_Admin_Plugin
+{
+
+    /**
+     * SVG image URL for the edit button.
+     */
+    const EDIT_IMAGE_URL = DOKU_URL . 'lib/plugins/findologicxmlexport/resources/edit.svg';
+
+    const EXPORT_URL = DOKU_URL . 'lib/plugins/findologicxmlexport';
+    const STYLESHEET_URL = DOKU_URL . 'lib/plugins/findologicxmlexport/resources/style.css';
+
+    /**
+     * Sort plugin in the DokuWiki admin interface.
+     * The lower this value is, the higher it is sorted.
+     */
+    const MENU_SORT = 1;
 
     /**
-     * @return int sort number in admin menu
+     * Template folder directory.
      */
-    public function getMenuSort() {
-        return 10000;
+    const TEMPLATE_DIR = __DIR__ . '/tpl';
+
+    /**
+     * Template file name. Directory is set in constant TEMPLATE_DIR.
+     */
+    const TEMPLATE_FILE = 'admin.tpl';
+
+    /**
+     * Variable name for translations set in the DokuWiki.
+     * It is used for the template.
+     */
+    const LANGUAGE_VARIABLE_NAME = 'languageText';
+
+    /**
+     * @return int sorting of the plugin in the plugin manager.
+     */
+    public function getMenuSort()
+    {
+        return self::MENU_SORT;
     }
 
     /**
-     * @return bool true if only access for superuser, false is for superusers and moderators
+     * Admins or higher can access this plugin.
+     *
+     * You do not need to be a superuser to access this plugin.
      */
-    public function forAdminOnly() {
+    public function forAdminOnly()
+    {
         return true;
     }
 
     /**
-     * Should carry out any processing required by the plugin.
+     * HTML output (gets generated by twig).
      */
-    public function handle() {
+    public function html()
+    {
+        $pagesWithoutTitle = PageGetter::getPagesWithoutTitle();
+
+        $urls = [
+            'editImageUrl' => self::EDIT_IMAGE_URL,
+            'exportUrl' => self::EXPORT_URL,
+            'stylesheetUrl' => self::STYLESHEET_URL
+        ];
+
+        $variablesForTemplate = ['var' => array_merge($pagesWithoutTitle, $urls)];
+
+        // Set up loader and environment for twig.
+        $loader = new Twig_Loader_Filesystem(self::TEMPLATE_DIR);
+        $twig = new Twig_Environment($loader);
+
+        $initLanguage = $this->getLang(''); // Loads the translations set in the plugin translation.
```

Comment:
* You can skip the return variable.
* Comment more on this (something about `getLang()` needs to be called once to initialize `$this->lang`).

Reply:
:white_check_mark: